packages/wrangler/src/sourcemap.ts

```diff
@@ -1,10 +1,60 @@
 import assert from "node:assert";
+import fs from "node:fs";
+import url from "node:url";
+import type { Options } from "@cspotcode/source-map-support";
 import type Protocol from "devtools-protocol";
 
+function maybeGetFile(filePath: string | URL) {
+	try {
+		return fs.readFileSync(filePath, "utf8");
+	} catch (e: unknown) {
+		const notFound =
+			typeof e === "object" && e !== null && "code" in e && e.code === "ENOENT";
+		if (!notFound) throw e;
+	}
+}
+
+export type RetrieveSourceMap = NonNullable<Options["retrieveSourceMap"]>;
+export function maybeRetrieveFileSourceMap(
+	filePath?: string
+): ReturnType<RetrieveSourceMap> {
+	if (filePath === undefined) return null;
+	const contents = maybeGetFile(filePath);
+	if (contents === undefined) return null;
+
+	// Find the last source mapping URL if any
+	const mapRegexp = /# sourceMappingURL=(.+)/g;
+	const matches = [...contents.matchAll(mapRegexp)];
+	// If we couldn't find a source mapping URL, there's nothing we can do
+	if (matches.length === 0) return null;
+	const mapMatch = matches[matches.length - 1];
+
+	// Get the source map
+	const fileUrl = url.pathToFileURL(filePath);
+	const mapUrl = new URL(mapMatch[1], fileUrl);
+	if (
+		mapUrl.protocol === "data:" &&
+		mapUrl.pathname.startsWith("application/json;base64,")
+	) {
+		// sourceMappingURL=data:application/json;base64,ew...
+		const base64 = mapUrl.href.substring(mapUrl.href.indexOf(",") + 1);
+		const map = Buffer.from(base64, "base64").toString();
+		return { map, url: fileUrl.href };
+	} else {
+		const map = maybeGetFile(mapUrl);
+		if (map === undefined) return null;
+		return { map, url: mapUrl.href };
+	}
+}
+
 let sourceMappingPrepareStackTrace: typeof Error.prepareStackTrace;
-function getSourceMappingPrepareStackTrace(): NonNullable<
-	typeof Error.prepareStackTrace
-> {
+let retrieveSourceMapOverride: RetrieveSourceMap | undefined;
```

Comment:
Why is this variable outside the `getSourceMappingPrepareStackTrace()` function? It doesn't seem to be used outside there?

Reply:
Ah I see. That was not at all clear from the code - for me at least 😅.
I think a more detailed comment at this declaration would help future readers of the code.